### DISCUSSION TITLE
記事内にQiita リンクカードが多い場合に、カードが表示されないことがある

### DISCUSSION THREAD

**Discussion by yoshi389111** [increments/qiita-discussions#226] [要望・アイデア・不具合]
記事内に [Qiita リンクカード](https://qiita.com/mziyut/items/bc00ed7396bd60a717b8#qiita-%E3%83%AA%E3%83%B3%E3%82%AF%E3%82%AB%E3%83%BC%E3%83%89) が数多くある場合（だと思うのですが）に、カードが表示されず、URLのままとなることがあります。

毎回発生するわけではなく、何度かリロードなどをすると表示されるようになることもあります（なので、クライアント側の処理で何かの制限がかかっているのかと思っています。タイムアウトとか、同時並行処理数とか）。

とある記事では 60 ～ 70 ぐらいの Qiita リンクカードがあり、30数個はカードが表示され、残りの30数個はURLのままとなっています（表示できる個数は変動します）。

たいていの場合には、上の方にあるものは正しくカードが表示されて、下にある方がURLのままのことが多いですが、必ずしも、上から順番に表示ができているわけではないようです。

![image](https://user-images.githubusercontent.com/15214393/169162477-5922acc7-7672-45c5-9a65-3f7f40c5be2d.png)

上の方にある緑色のリンクが、カードにならずにURLとなっているものです。

![image](https://user-images.githubusercontent.com/15214393/169162715-6f69e278-9889-43eb-b1a1-b402fa73bdd1.png)

リロードすると、状況が変わります（今回の場合には、上の緑色のリンクの URL がカードになった代わりに、その下のカードが URL になってしまっている）。

１ページ内の Qiita リンクカードの上限数とかがあるのでしょうか？

* WIndows 12
* Chrome 101.0.4951.67

**Comment by tanaka-asami415** [comment]
@yoshi389111 
お問い合わせありがとうございます🙏

数の上限数はありません。我々で確認してみたところ再現ができなかったのですが、現在も発生しておられるでしょうか？

もしよろしければ、発生時の動作を一度見せてもらえたら…と思っているのですが、
機会をいただくことは可能でしょうか？🙇

**Reply by yoshi389111** [reply]
たとえば、この [ページ](https://qiita.com/yoshi389111/items/6ba5966f739ea01c6719) で起きるのですが、正しく全部が表示される時と、表示されないときがあります。

感覚的には、昼間は全部カード表示されやすくて、夜間や早朝は後半がカードにならないことが多いイメージですが、あまり根拠はないです。
もちろん昼間でもカードにならないことはあります。今も再現しました。

（もしかしたら、夜間に起きやすいというのは、こちらのネットワークの混雑具合などが原因かもしれないです）

**Reply by tanaka-asami415** [reply]
@yoshi389111 
根本的な解決になるかはわからないのですが
若干これが影響しているかも？という部分を修正してみました！

もし、まだ同じような問題があれば教えていただけると幸いです！🙏

**Reply by yoshi389111** [reply]
ありがとうございます。様子を見てみます。

**Reply by yoshi389111** [reply]
その後、特にQiitaリンクカードが展開されない事象は起きていないです。

ちょっと気になるのは、以下の2点です。

* Qiitaリンクカードが表示領域になった直後に切り替わるようになっているので、ちらつくようでちょっと気になる
* スクロールバーをドラッグして記事の最後の方にスクロールしようとすると、途中で記事全体の長さが伸びるためか、１回では最後までスクロールできない

大した問題ではないですが。

対応、ありがとうございました。

お手数ですが、本件はクロースしていただけるでしょうか。よろしくお願いします。

**Reply by tanaka-asami415** [reply]
@yoshi389111 
ご連絡ありがとうございます！

ご指摘いただいた内容もなんとかしたいなと思ってはいますが、他の優先度など踏まえすぐに対応ができなさそうです…
気長にお待ちいただけると幸いです🙏

本件、一旦クローズさせていただきます🙇
また何かありましたら、discussionお待ちしております！
